install/helm/agones/templates/crds/_gameserverspecvalidation.yaml

```diff
@@ -92,6 +92,17 @@ properties:
               type: integer
               minimum: 1
               maximum: 65535
+      sdkServerLogLevel:
```

Comment:
You use sdk server in some places and game server sidecar in others. I'm not sure which is better, but we should be consistent one way or the other. I'm a bit partial to game server sidecar, since users of agones won't realize that the sidecar has to do with the SDK and because when you are looking at the logs in a k8s cluster those are the words used to name the container we inject into the pod with the game server container. 

I'd suggest making this field be `gameserverSidecarLogLevel`.

Reply:
So this took me back - I didn't realise that was what we called the sdk server sidecar container!
`"agones-gameserver-sidecar"` - in retrospect, possibly not the best name! But agree, consistency is key.

I guess my only question is - do we ever thing we'd add another sidecar of some sort? In which case, should we be specific about which sidecar's log level we are manipulating?

Reply:
> I guess my only question is - do we ever thing we'd add another sidecar of some sort? 

Are you suggesting making it `sidecarLogLevel` for now (shorter) or are you saying that we might have another sidecar for gameservers and that `gameserverSidecarLogLevel` isn't specific enough?

Reply:
Maybe the gameserver prefix is unnecessary?

Reply:
My concern was that "sidecar" could be too generic - that's why I was thinking sdk-server.

My other concern would be that "sidecar" is a well known K8s pattern, but I'm not sure if its knowledge we can assume across all our users?

Reply:
I'd suggest either `agonesGameserverSidecarLogLevel` (if we want to exactly match the name of the sidecar container) or `gameserverSidecarLogLevel` (if we want a shorter field name). I'm inclined to use the longer name if we aren't going to shorten it all the way to "sidecar".

Reply:
I think `agones` is redundant, since we're on an Agones GameServer. Ditto for `gameserverSidecarLogLevel` - we are in a gameServer already.

Maybe I wasn't clear - I prefer `sdkServerLogLevel` because it's specific to which thing it is being set to. "sidecar" could essentially be anything, and is also confusing to those who don't know Kubernetes patterns. What if the user creates their own sidecars, are we somehow overwriting their log level? If we ever add another sidecar for Agones GameServers (not likely, but possible), now we can't configure log levels for each one individually, because we don't have a name for each one.

Does that make more sense?

Reply:
I also like sdkServerLogLevel more.
However I can change parameter to gameserverSidecarLogLevel if it would be more natural to the users.

Reply:
I had an idea this morning in the shower :shower: 

I think we should have a logging section. Looking at commentary in #1036 , it's possible we will have more than 1 sidecar. In which case, we likely don't want multiple top spec level logging config params.

So howabout this:

```yaml
spec:
  # ... other fields ... 
  logging:
    sdkServer: "warning"
```

Then if we have to add more logging parameters, they aren't peppered through the config, they can be scoped under this `logging` element.

WDYT?

Reply:
Or maybe that should be `loggingLevel` ? (or `level` should be a sub-element to `logging` ?) Not 100% sure. But I think this is on a better path.